lib/srv/desktop/rdp/rdpclient/src/rdpdr/mod.rs

```diff
@@ -252,256 +258,260 @@ impl Client {
+                                return cli.tdp_sd_create(rdp_req, FileType::Directory, res.fso);
+                            } else {
+                                // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L258
+                                // ERROR_FILE_NOT_FOUND --> STATUS_NO_SUCH_FILE: https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_main.c#L85
+                                return cli.prep_device_create_response(
+                                    &rdp_req,
+                                    NTSTATUS::STATUS_NO_SUCH_FILE,
+                                    0,
+                                );
+                            }
+                        }
                     } else {
-                        self.prep_device_close_response(rdp_req, NTSTATUS::STATUS_SUCCESS)
+                        // generic RDP error
+                        return cli.prep_device_create_response(
+                            &rdp_req,
+                            NTSTATUS::STATUS_UNSUCCESSFUL,
+                            0,
+                        );
                     }
-                } else {
-                    self.prep_device_close_response(rdp_req, NTSTATUS::STATUS_UNSUCCESSFUL)
-                }
+
+                    // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L263
+                    // Note: the above line in FreeRDP will always evaluate to true:
+                    // - drive_file_init is called here: https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L362
+                    // - it is always passed a file where file->file_handle = INVALID_HANDLE_VALUE: https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/channels/drive/client/drive_file.c#L351
+                    // - None of the calls up to the line in question can have changed it
+
+                    // The actual creation of files and error mapping badevice_io_request.completion_id happens here, for reference:
+                    // https://github.com/FreeRDP/FreeRDP/blob/511444a65e7aa2f537c5e531fa68157a50c1bd4d/winpr/libwinpr/file/file.c#L781
+                    if rdp_req.create_disposition == flags::CreateDisposition::FILE_SUPERSEDE {
+                        // If the file already exists, replace it with the given file. If it does not, create the given file.
+                    } else if rdp_req.create_disposition == flags::CreateDisposition::FILE_OPEN {
+                        // If the file already exists, open it instead of creating a new file. If it does not, fail the request and do not create a new file.
+                        if res.err_code == TdpErrCode::Nil {
```

Comment:
nit: Do we have any preference in our code base to use either `match` statements or `if / else` statements?

Reply:
I tend to prefer match over a large if/else chain